staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -60,14 +63,57 @@ type ResourceManager interface {
 	// Thread-safe
 	RemoveGroupVersion(gv metav1.GroupVersion)
 
-	// Resets the manager's known list of group-versions and replaces them
-	// with the given groups
-	// Thread-Safe
 	SetGroups([]apidiscoveryv2beta1.APIGroupDiscovery)
 
+	// Returns the same resource manager using a different source
+	// The source is used to decide how to de-duplicate groups.
+	// DefaultSource always takes precedence. Otherwise, the precedence between
+	// two sources is not defined.
```

Comment:
Should we enforce that the source should always be default or aggregator? Ideally we should try to avoid undefined behavior